Automattic/wp-calypso
# Login: Hide password field until user submits email or username

This pull request is another iteration after https://github.com/Automattic/wp-calypso/pull/19277 aiming at updating the [`Login` page](https://wordpress.com/log-in) and any related page to better support **passwordless accounts**.

Here the goal is simply to hide the password field when the `Login` page loads, and show it once the user has submitted the form with an email address or a username. In the future, we will check the type of authentication of the account via a new API endpoint when the user clicks the `Continue` button, and either display the password field for regular accounts, or send an email with a magic link for passwordless accounts. You can find the mockup of this new flow at p6qnuF-5T0-p2#comment-6882.

##### After Page Load

<img width="456" alt="screenshot" src="https://user-images.githubusercontent.com/594356/32700126-f0b3a1e8-c7c0-11e7-9d70-bcabbf6122cd.png">

##### After Submit

<img width="456" alt="screenshot" src="https://user-images.githubusercontent.com/594356/32700131-0105924a-c7c1-11e7-8c8f-75723a04d3e0.png">

##### When Linking Account

<img width="456" alt="screenshot" src="https://user-images.githubusercontent.com/594356/32701046-30e734b0-c7cf-11e7-89e9-11a623662c5c.png">

#### Testing instructions

You can check that the new Track events are correctly triggered by executing the following command in your browser's console, and then reloading the page:

```
localStorage.setItem( 'debug', 'calypso:analytics:tracks' );
```

1. Run `git checkout update/login-flow` and start your server, or open a [live branch](https://calypso.live/?branch=update/login-flow)

##### Default login

2. Open the [`Login` page](http://calypso.localhost:3000/log-in) in an incognito window
3. Assert that the `Password` field is hidden
4. Assert that the label of the submit button is `Continue`
5. Fill in the `Email Address or Username` field, and click this `Continue` button
6. Assert that the `Password` field is now displayed, and has focus
7. Assert that the label of the submit button is now `Log In`
8. Assert that you can still log in

##### Default login (linking flow)

2. Log in with a WordPress.com account that is already connected to Google
3. Navigate to the [`Social Login` page](http://calypso.localhost:3000/social-login)
4. Click the `Disconnect` button
5. Sign out from WordPress.com
6. Open the [`Login` page](http://calypso.localhost:3000/log-in)
7. Click the `Continue with Google` button
8. Authenticate with Google
9. Assert that a `We found a WordPress.com account with the email address [...]` message is shown
10. Assert that the `Password` field is displayed, and has focus
11. Assert that the label of the submit button is `Log In`
12. Assert that you can still log in

##### WooCommerce

2. Open the WooCommerce [`Home` page](https://woocommerce.com/) in an incognito window
3. Click the `SIGN IN WITH WORDPRESS.COM` link in the header
4. Replace https://wordpress.com with http://calypso.localhost:3000 in the address bar, and reload
5. Follow steps `#3` to `#8` of the default login

##### Other OAuth apps

2. Open the Akismet [`Home` page](https://akismet.com/) in an incognito window
3. Click the `Sign In` button in the header
4. Replace https://wordpress.com with http://calypso.localhost:3000 in the address bar, and reload
5. Follow steps `#3` to `#8` of the default login

#### Reviews
